src/Services/CRM/Address/Service/Address.php

```diff
@@ -0,0 +1,239 @@
+ */
+
+declare(strict_types=1);
+
+namespace Bitrix24\SDK\Services\CRM\Address\Service;
+
+use Bitrix24\SDK\Attributes\ApiEndpointMetadata;
+use Bitrix24\SDK\Attributes\ApiServiceMetadata;
+use Bitrix24\SDK\Core\Contracts\CoreInterface;
+use Bitrix24\SDK\Core\Credentials\Scope;
+use Bitrix24\SDK\Core\Exceptions\BaseException;
+use Bitrix24\SDK\Core\Exceptions\TransportException;
+use Bitrix24\SDK\Core\Result\AddedItemResult;
+use Bitrix24\SDK\Core\Result\DeletedItemResult;
+use Bitrix24\SDK\Core\Result\FieldsResult;
+use Bitrix24\SDK\Core\Result\UpdatedItemResult;
+use Bitrix24\SDK\Services\AbstractService;
+use Bitrix24\SDK\Services\CRM\Address\Result\AddressResult;
+use Bitrix24\SDK\Services\CRM\Address\Result\AddressesResult;
+use Psr\Log\LoggerInterface;
+#[ApiServiceMetadata(new Scope(['crm']))]
+class Address extends AbstractService
+{
+    public Batch $batch;
+
+    /**
+     * Address constructor.
+     *
+     * @param Batch           $batch
+     * @param CoreInterface   $core
+     * @param LoggerInterface $log
+     */
+    public function __construct(Batch $batch, CoreInterface $core, LoggerInterface $log)
+    {
+        parent::__construct($core, $log);
+        $this->batch = $batch;
+    }
+
+    /**
+     * add new address
+     *
+     * @link https://apidocs.bitrix24.com/api-reference/crm/requisites/addresses/crm-address-add.html
+     *
+     * @param array{
+     *   TYPE_ID?: int,
+     *   ENTITY_TYPE_ID?: int,
+     *   ENTITY_ID?: string,
+     *   ADDRESS_1?: string,
+     *   ADDRESS_2?: string,
+     *   CITY?: string,
+     *   POSTAL_CODE?: string,
+     *   REGION?: string,
+     *   PROVINCE?: string,
+     *   COUNTRY?: string,
+     *   COUNTRY_CODE?: string,
+     *   LOC_ADDR_ID?: int,
+     *   } $fields
+     *
+     * @return AddedItemResult
+     * @throws BaseException
+     * @throws TransportException
+     */
+    #[ApiEndpointMetadata(
+        'crm.address.add',
+        'https://apidocs.bitrix24.com/api-reference/crm/requisites/addresses/crm-address-add.html',
+        'Method adds new address'
+    )]
+    public function add(array $fields): AddedItemResult
+    {
+        return new AddedItemResult(
+            $this->core->call(
+                'crm.address.add',
+                [
+                    'fields' => $fields,
+                ]
+            )
+        );
+    }
+
+    /**
+     * Deletes the specified address.
+     *
+     * @link https://apidocs.bitrix24.com/api-reference/crm/requisites/addresses/crm-address-delete.html
+     *
+     * @param int $id
+     *
+     * @return DeletedItemResult
+     * @throws BaseException
+     * @throws TransportException
+     */
+    #[ApiEndpointMetadata(
+        'crm.address.delete',
+        'https://apidocs.bitrix24.com/api-reference/crm/requisites/addresses/crm-address-delete.html',
+        'Deletes the specified address.'
+    )]
+    public function delete(array $fields): DeletedItemResult
```

Comment:
i think we can split $fields to typed and annotated parameters

https://apidocs.bitrix24.com/api-reference/crm/requisites/addresses/crm-address-delete.html